Remove fields that are no longer in the `buf.lock` (#80)
bufbuild/docs.buf.build: docs/configuration/v1/buf-lock.md

  - remote: buf.build
    owner: acme
    repository: paymentapis
    commit: 5173e5cfeb904508839378050d95e1de
```

> Older versions of `buf` may include `branch`, `commit`, `digest`, and `create_time`

> > as a part of the dependencies. The latest `buf.lock` file will not include these fields when

Reviewer: I'd rephrase this second sentence:

"Your `buf.lock` shouldn't include these fields if you've run `buf mod update` with a newer version of `buf`."

> Note that older versions of `buf` may include `branch`, `commit`, `digest`, and `create_time`
> as a part of the dependencies. Older `buf.lock` files are updated without the fields when
> `buf mod update` is run on a newer version of `buf`.

Reviewer 2: Let's include this as a block in the note format we use elsewhere.
```suggestion
> Older versions of `buf` may include `branch`, `commit`, `digest`, and `create_time`
> as a part of the dependencies. The latest `buf.lock` file will not include these fields when
> `buf mod update` is run on a newer version of `buf`.
```

We can land this as-is, but flagging @lucperkins for potential language updates in a follow up (if ever).
